### PR TITLE
PORTALS-1706

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/lib/containers/RssFeedCards.tsx [hunk 1]
```diff
@@ -35,13 +40,18 @@ export default class RssFeedCards extends React.Component<RssFeedCardsProps, Rss
   
   componentDidMount() {
     this._isMounted = true
-    const { url } = this.props
-    fetch(url)
+    const { url, lockedFacet } = this.props
+    const lockedFacetValue = lockedFacet?.value
+    const tagPath = lockedFacetValue ? `/tag/${lockedFacetValue.replace(' ', '-')}` : ''
```
Comment (jay-hodgson) [inline src/lib/containers/RssFeedCards.tsx:45]: If a lockedFacet is provided, then the value is used as a tag to filter the items shown.

File: src/lib/containers/RssFeedCards.tsx [hunk 2]
```diff
@@ -60,6 +70,18 @@ export default class RssFeedCards extends React.Component<RssFeedCardsProps, Rss
   }
 
   render() {
+    const { viewAllNewsButtonText } = this.props
+    
+    if (this.state.rssFeed.items?.length === 0) {
```
Comment (jay-hodgson) [inline src/lib/containers/RssFeedCards.tsx:75]: Add UI to show when no items are available.

File: src/lib/containers/RssFeedCards.tsx [hunk 3]
```diff
@@ -130,10 +152,10 @@ export default class RssFeedCards extends React.Component<RssFeedCardsProps, Rss
             })}
           </div>
           {this.state.rssFeed.items &&
-            this.state.rssFeed.items.length > this.props.itemsToShow && (
+            this.state.rssFeed.items.length > this.props.itemsToShow && this.state.allItemsUrl && (
               <div className="RssFeedViewAllNewsButtonContainer">
-                <a className="homepage-button-link" href={this.state.rssFeed.link} target="_blank" rel="noopener noreferrer">
-                    VIEW ALL NEWS
+                <a className="homepage-button-link" href={this.state.allItemsUrl} target="_blank" rel="noopener noreferrer">
+                    { viewAllNewsButtonText ?? 'VIEW ALL NEWS' }
```
Comment (jay-hodgson) [inline src/lib/containers/RssFeedCards.tsx:158]: Ability to override "View All News" button text.

**Review by emmanmills — APPROVED** [review]
Looks good to me!

**Pull request merged**